### PR TITLE
validate settings payload

### PR DESCRIPTION
**Pull request by sickelap**
- add validation to settings payload using `jsonschema`
- accept (validated) value for heavyweight_process

related PR https://github.com/LibrePhotos/librephotos-frontend/pull/138

### CODE REVIEW

File: api/views/views.py [hunk 1]
```diff
@@ -81,23 +83,15 @@ def get(self, request, format=None):
...
         if "heavyweight_process" in request.data.keys():
-            HEAVYWEIGHT_PROCESS_ENV = request.data["heavyweight_process"]
-            HEAVYWEIGHT_PROCESS = (
-                int(HEAVYWEIGHT_PROCESS_ENV)
-                if HEAVYWEIGHT_PROCESS_ENV.isnumeric()
-                else 1
-            )
-            site_config.HEAVYWEIGHT_PROCESS = HEAVYWEIGHT_PROCESS
-        if "map_api_key" in request.data.keys():
-            site_config.MAP_API_KEY = request.data["map_api_key"]
```
Comment (derneuere) [inline api/views/views.py:99]: I think you lost map_api_key, when implementing the validating logic

Comment (sickelap) [reply api/views/views.py:99]: uh, good catch! I'll fix it this evening.

Comment (sickelap) [reply api/views/views.py:99]: one more reason to start thinking of automated tests

**Comment by derneuere** [comment]
Looks good to me 👍

**Comment by DigitalWarrior** [comment]
Since this commit dev branch does now allow scanning. Verified on demo2.librephtos.com
Resetting the scan directory does not fix the issue. 

image
and a users instance
image
@sickelap @derneuere

**Comment by derneuere** [comment]
@sickelap The issue is in this commit here: https://github.com/LibrePhotos/librephotos-frontend/commit/e2a80258651b832006821e7fd32e8ce483b1d035

I think "buttonDisabled" was only used for debugging purposes, as it was set to false all the time. Changing it to workerAvailability makes sense, but it does not seem to fetch it correctly. Could you look into it?

**Comment by sickelap** [comment]
I already had a chat with @DigitalWarrior and I know where the problem is. I'll be fixing this now.

**Pull request merged**